src/main/java/com/linkedin/camus/etl/kafka/coders/ProtobufMessageDecoder.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.camus.etl.kafka.coders;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import com.linkedin.camus.coders.CamusWrapper;
+import com.linkedin.camus.coders.MessageDecoder;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.InvalidParameterException;
+import java.time.Instant;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+public class ProtobufMessageDecoder extends MessageDecoder<byte[], Message> {
+    public static final String PROTOBUF_MESSAGE_CLASS_PREFIX = "camus.protobuf.message.class.";
+    public static final String CAMUS_MESSAGE_TIMESTAMP_FIELD = "camus.message.timestamp.field";
+    public static final String DEFAULT_TIMESTAMP_FIELD = "ts";
+
+    private Method messageParseMethod;
+    private String timestampFieldName;
+
+    @Override
+    public void init(Properties props, String topicName) {
+        super.init(props, topicName);
+
+        String messageClassForTopic = props.getProperty(PROTOBUF_MESSAGE_CLASS_PREFIX + topicName);
```

Comment:
As each Camus job is per-topic with its own set of properties, maybe it would be simpler to have a property called "camus.protobuf.message.class" rather than include the topic name?

Reply:
I'd say it's easier to configure the application with a *.properties file and specify the association between topics and classes there:

```
camus.protobuf.message.class.attributions=events.AttributionEventOuterClass$AttributionEvent
camus.protobuf.message.class.bids=events.BidEventOuterClass$BidEvent
camus.protobuf.message.class.clicks=events.BidEventOuterClass$BidEvent
camus.protobuf.message.class.losses=events.LossEventOuterClass$LossEvent
camus.protobuf.message.class.trigger_results=trigger.TaskOuterClass$TaskResult
```

Otherwise, we need to put this internal knowledge into the Chef's recipe in order to be able to specify it as a command-line argument. Moreover, the Secor config looks the same, so it's a kind of unification between applications, which is also good :)